### ISSUE TITLE
Just a thought: Use modern AI for reconciling transactions?

### ISSUE THREAD

**Issue by bjendres** [Project60/org.project60.banking#417]
In theory it _should_ make sense to use a modern AI model to further automate your CiviBanking configuration.

Here are some general thoughts about this:

1. It would _have to_ use a private, secure, local model, since CiviBanking processes loads of personalised and financial data
2. The input data is well defined by the fields in the bank transaction entity, and so is the outcome.
3. If you've been using CIviBanking for a while, you already have the perfect training data for that model: all your past reconciliation decisions (automatic/manual) are still recorded in the old bank transaction entities. Alternatively you could have a 'passive' mode, so it would learn while you do the conventional reconciliation.
4. Analogously to the coded/configured matchers, the module could create suggestions, or automatically reconcile if it's sufficiently confident.

Discuss! :-)

**Comment by bjendres** [comment]
These are all valid points, but you seem be working under the assumption that the human "processor" doesn't make any mistakes -  that is specifically not true if they have to go through hundreds of transactions at a time manually.

But I agree that this, should it ever happen, should be a niche thing for specific scenarios and not the standard setup, probably for large volumes that cannot be easily captured by a handful of rules.

**Comment by jensschuppe** [comment]
No, I do not assume humans do not make mistakes. It is about responsibility in case of errors. Also, a human might be able to remember circumstances when something seems weird after a wrong decision. An AI cannot provide that contextual memory. Also, the model would not be able to learn from such mistakes as there is no feedback for manual corrections (by humans).

I might be the devil's advocate here, but AI does not seem to be the right tool for decision-making at all. It might provide more aspects for an analysis, as it knows about known data, but it does not have implicit knowledge like humans do.
